### PR TITLE
Android support decode scale

### PR DESCRIPTION
**Pull request by WizzXu**
encoded AVIF image size while be
  decode with size of bitmap

### CODE REVIEW

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (wantehchang) [inline android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:150]: Vignesh: I suggest we remove the `diag` parameter from the `avifImageScale()` function first.

Until that is done, this pull request should declare a local `avifDiagnostics diag` variable and pass `&diag` to `avifImageScale()`.

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/jni/libavif_jni.cc]: I have modified this problem

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/jni/libavif_jni.cc]: I have modified this problem

File: android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java
Comment (vigneshvg) [inline android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java:119]: couple of points:
1) This is an API breaking change. So it should be done as a new overloaded function with false being passed from the existing ones to keep backwards compatibility.

2) i don't think the `isScale` parameter is necessary at all. we could just update the function comments to say that if the `bitmap` dimensions do not match the decoded image's dimensions, then the decoded image will be scaled to match the bitmap's dimensions (i.e.) scaling could be the default behavior.

Comment (vigneshvg) [inline android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java:119]: We cannot change the behavior of this function alone without updating the behaviors of `nextFrame` and `nthFrame` functions as well.

However, adding scaling support to those functions might involve making a copy of the image in the native code since we may not be able to scale the decoder's copy of the image directly.

If you want this PR to be merged as-is, that's probably ok ,but can you please add a TODO to support scaling on `nthFrame` and the `nextFrame` functions.

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (vigneshvg) [inline android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:194]: Duplicating this function is not necessary. You can simply pass true/false from each caller instead.

File: android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java
Comment (vigneshvg) [inline android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java:176]: could you please add more cases?
* larger bitmap,
* bitmap with different aspect ratios than the source image

**Review by vigneshvg — CHANGES_REQUESTED** [review]

**Comment by vigneshvg** [comment]
Thank you for working on this PR. :) I have a few comments.

File: android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java
Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java:119]: For the first point, I will modify it
For the second point, what I start to think is not affecting the original API. If you join this feature uniformly, I think it can be removed

Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/java/org/aomedia/avif/android/AvifDecoder.java:119]: I talk about removing the `isScal` parameter. As the default function, it will support `nextFrame` and `nthFrame `

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:194]: yes

File: android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java
Comment (WizzXu) [reply android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java:176]: yes

**Review by vigneshvg — CHANGES_REQUESTED** [review]

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (WizzXu) [reply android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:140]: I'm sorry, I'm not able to do this change, could you please add a commit to this PR

Comment (vigneshvg) [reply android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:140]: yes i will update the PR.

File: android_jni/avifandroidjni/src/main/jni/libavif_jni.cc
Comment (wantehchang) [inline android_jni/avifandroidjni/src/main/jni/libavif_jni.cc:149]: Nit: Use `%s` with `avifResultToString(res)`, or print both?

Also at line 157.

File: android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java
Comment (wantehchang) [inline android_jni/avifandroidjni/src/androidTest/java/org/aomedia/avif/android/AvifDecoderTest.java:228]: Nit: have => has

**Review by wantehchang — APPROVED** [review]
LGTM. I only reviewed the last commit by Vignesh.

**Review by vigneshvg — APPROVED** [review]

**Comment by vigneshvg** [comment]
I am looking into why the emulator tests are failing on the CI. I am not able to get them to fail locally.

**Comment by vigneshvg** [comment]
> I am looking into why the emulator tests are failing on the CI. I am not able to get them to fail locally.

The Android Emulator CI seems to be broken in the current head (without this PR). Looking into it.

**Comment by vigneshvg** [comment]
> > I am looking into why the emulator tests are failing on the CI. I am not able to get them to fail locally.
> 
> The Android Emulator CI seems to be broken in the current head (without this PR). Looking into it.

The CI will pass once #1704 is merged. I will merge this PR once that is merged and the CI passes.

**Comment by vigneshvg** [comment]
> > > I am looking into why the emulator tests are failing on the CI. I am not able to get them to fail locally.
> > 
> > 
> > The Android Emulator CI seems to be broken in the current head (without this PR). Looking into it.
> 
> The CI will pass once #1704 is merged. I will merge this PR once that is merged and the CI passes.

This PR is now merged.

@WizzXu thank you again for working on this PR.

**Pull request merged**
